system/jlib/jfile.cpp

```diff
@@ -921,6 +921,16 @@ void CFile::setReadOnly(bool ro)
 #endif
 }
 
+void CFile::setFilePermissions(unsigned fPerms)
+{
+#ifndef _WIN32
+    struct stat info;
+    if (stat(filename, &info) != 0)
+        throw makeErrnoExceptionV("CFile::setFilePermissions() %s", filename.get());
+    chmod(filename, fPerms&0777);
```

Comment:
related to previous remote side comment I suppose.
setFilePermissions() could return a boolean to indicate if chmod succeeded or not.
(same true of setReadOnly() too really)